packages/twenty-server/src/engine/twenty-orm/factories/workspace-datasource.factory.ts

```diff
@@ -355,8 +365,37 @@ export class WorkspaceDatasourceFactory {
   }
 
   public async destroy(workspaceId: string) {
-    await this.promiseMemoizer.clearKeys(`${workspaceId}-`, (dataSource) => {
-      dataSource.destroy();
-    });
+    try {
+      await this.promiseMemoizer.clearKeys(
+        `${workspaceId}-`,
+        async (dataSource) => {
+          try {
+            await dataSource.destroy();
+          } catch (error) {
+            // Ignore known race-condition errors to prevent noise during shutdown
+            if (
+              error.message === 'Called end on pool more than once' ||
+              error.message?.includes(
+                'pool is draining and cannot accommodate new clients',
+              )
+            ) {
+              this.logger.debug(
+                `Ignoring pool error during cleanup: ${error.message}`,
+              );
+
+              return;
+            }
+
+            // Re-throw unexpected errors so they are handled by the outer catch
+            throw error;
+          }
+        },
+      );
+    } catch (error) {
+      // Log and swallow any errors during cleanup to prevent crashes
```

Comment:
I'm a bit lost, why do we have two try catch nested here, can we avoid it? 
+ does not seems to be a good idea to swallow all errors, they should go to sentry. If there is a specific exception which is a known case that we want so swallow, we should swallow only this one but let the other flow. And in this case, we should swallow them as close as possible from their known trigger point in the call stack